src/cljc/markdown/transformers.cljc

```diff
@@ -125,9 +125,9 @@
     (str " " text) text))
 
 (defn paragraph
-  [text {:keys [eof heading inline-heading temp hr code lists blockquote paragraph last-line-empty?] :as state}]
+  [text {:keys [eof heading inline-heading temp hr code codeblock lists blockquote paragraph last-line-empty? end-codeblock?] :as state}]
   (cond
-    (and paragraph lists)
+    (or (and paragraph lists) (and paragraph codeblock))
```

Comment:
Yes I agree it should get split up. My pull request makes the previous example I gave give this:
```
user=> (md-to-html-string "hey\n```\nhey\n```\nhey\nhey")
"<p>hey</p><pre><code>hey\n</code></pre><p>hey hey</p>"
```